.github/workflows/widget-cd.yml

```diff
@@ -71,16 +71,34 @@ jobs:
           CONTEXT_PATH: ${{env.APP_CONTEXT_PATH}}
           DELETE_EXISTING: false
 
-  e2e-tests:
+  e2e-tests-manual:
+    if: ${{ github.event_name == 'workflow_dispatch' }}
     needs: [deploy]
     uses: ./.github/workflows/cypress-test.yml
     with:
-      tenantUrl: ${{ needs.deploy.determine-target-tenant.outputs.tenantUrl }}
+      tenantUrl: ${{ github.event.inputs.tenant_url }}
```

Comment:
Separate the tenant determining logic into a separate job. Add it in the needs section. Follow setup-variables job in our workflows. Have only one step for e2e tests

Reply:
That will not work please refer [4f78f67](https://github.com/SoftwareAG/cumulocity-asset-properties-widget/pull/27/commits/4f78f67aac4ca8dd1c99f74b964242b5e2f2a524)

Reply:
I have updated the code please review